src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -271,15 +280,34 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             return declarator != null;
         }
 
-        private static ExpressionSyntax GetNullCheckOperand(ExpressionSyntax left, ExpressionSyntax right)
+        private static ExpressionSyntax GetNullCheckOperand(ExpressionSyntax left, SyntaxNode right)
         {
             if (left.IsKind(SyntaxKind.NullLiteralExpression))
             {
-                return right;
+                // null == x
+                // null != x
+                return (ExpressionSyntax)right;
             }
 
             if (right.IsKind(SyntaxKind.NullLiteralExpression))
             {
+                // x == null
+                // x != null
+                return left;
+            }
+
+            if (right.IsKind(SyntaxKind.PredefinedType, out PredefinedTypeSyntax predefinedType)
+                && predefinedType.Keyword.IsKind(SyntaxKind.ObjectKeyword))
```

Comment:
what happens if someone writes `o == object`?  I suppose we just don't care about that sort of odd case?

Reply:
No idea. I'll think about ways to improve this.

Reply:
Added checks for the comparison kind